packages/playwright-core/src/server/dispatchers/androidDispatcher.ts

```diff
@@ -170,10 +172,33 @@ export class AndroidDeviceDispatcher extends Dispatcher<AndroidDevice, channels.
   }
 }
 
-export class AndroidSocketDispatcher extends Dispatcher<SocketBackend, channels.AndroidSocketChannel, AndroidDeviceDispatcher> implements channels.AndroidSocketChannel {
+class SocketSdkObject extends SdkObject implements SocketBackend {
+  private _socket: SocketBackend;
+  private _eventListeners;
+
+  constructor(parent: SdkObject, socket: SocketBackend) {
+    super(parent, 'socket');
+    this._socket = socket;
+    this._eventListeners = [
+      eventsHelper.addEventListener(socket, 'data', data => this.emit('data', data)),
+      eventsHelper.addEventListener(socket, 'close', () => this.emit('close')),
+    ];
+  }
+
+  async write(data: Buffer) {
+    await this._socket.write(data);
+  }
+
+  close() {
+    this._socket.close();
+    eventsHelper.removeEventListeners(this._eventListeners);
```

Comment:
This causes `DEBUG=pw:api npm run atest android/device.spec.ts:26` to fail:
1. `SocketSdkObject.close()` gets called
2. `'close'` event listeners get immediately removed
3. The client never receives the actual `'close'` event.